controllers/machinedeletionremediation_controller.go

```diff
@@ -138,16 +152,18 @@ func (r *MachineDeletionRemediationReconciler) SetupWithManager(mgr ctrl.Manager
 		Complete(r)
 }
 
-func (r *MachineDeletionRemediationReconciler) getRemediation(ctx context.Context, req ctrl.Request) *v1alpha1.MachineDeletionRemediation {
+func (r *MachineDeletionRemediationReconciler) getRemediation(ctx context.Context, req ctrl.Request) (*v1alpha1.MachineDeletionRemediation, error) {
 	remediation := new(v1alpha1.MachineDeletionRemediation)
 	key := client.ObjectKey{Name: req.Name, Namespace: req.Namespace}
 	if err := r.Client.Get(ctx, key, remediation); err != nil {
-		if !errors.IsNotFound(err) {
-			r.Log.Error(err, "error retrieving remediation in namespace", "remediation name", req.Name, "namespace", req.Namespace)
+		if apiErrors.IsNotFound(err) {
+			r.Log.Info("MDR already deleted, nothing to do")
+			return nil, nil
 		}
-		return nil
+		r.Log.Error(err, "error retrieving remediation in namespace", "remediation name", req.Name, "namespace", req.Namespace)
```

Comment:
`error retrieving remediation in namespace` -> `error retrieving a remediation object in namespace`